lib/default.nix

```diff
@@ -168,4 +168,33 @@ in {
         gomod2nix --outdir "$CURDIR"
       '';
     };
+  # `hermesModuleConfigToml { modules }).config.hermes.toml`
+  # will be a string to put into [config.toml](https://hermes.informal.systems/documentation/configuration/configure-hermes.html)
+  hermesModuleConfigToml = {modules}:
+    pkgs.lib.evalModules {
+      modules =
+        [
+          (
+            {
+              lib,
+              config,
+              ...
+            }: let
+              prev = config.hermes;
+              # remove `null` from toml render
+              cfg = prev // {chains = builtins.map (pkgs.lib.filterAttrsRecursive (_: v: v != null)) prev.chains;};
+              hermes-toml = nix-std.lib.serde.toTOML (builtins.removeAttrs cfg ["toml"]);
+            in
+              with lib; {
+                options.hermes =
+                  (import ../nixosModules/hermes/options.nix {inherit lib;})
+                  // {
+                    toml = mkOption {type = types.unique {message = "only one toml output";} types.str;};
+                  };
+                config.hermes.toml = hermes-toml;
+              }
+          )
```

Comment:
best explanation. how nix validates types and contstraints on data and allows users to provide their configurable things and override and share configs amid other things? that is called modules. there is no well know option i know.

Reply:
so basically if you want to type/validate/default/compose some config (like reuse port for hermese and for chain node, etc) - use modules.

Reply:
but what if we do not run modules, but some other things? just some adhoc config?

we still can use modules as cfg, like terranix does. so this approach is 100% like terranix one.

Reply:
like here https://github.com/ComposableFi/composable/blob/main/flake/cosmos/hermes.nix

Reply:
tbh I think it is like poor man good solution, because not ALL runners not all PACKAGES has MODULES.

If all packages and all runners would have OPTIONS/CFG modules, than i doubt this would be needed. but there are dozens of runners and not only cosmos nodes.

Reply:
> https://github.com/ComposableFi/composable/blob/main/flake/cosmos/hermes.nix

so one can provide any OPTIONS/CFG modules, and get ALL configs if he wants, but one must be HERMES